src/main/java/present/Search.java

```diff
@@ -15,6 +23,78 @@
    * @return the {@code n} nearest neighborhoods, ordered nearest to farthest
    */
   public static List<Neighborhood> near(Location location, int n) {
-    throw new UnsupportedOperationException("Please implement!");
+    Map<Neighborhood, Double> sortedNeighborhoodMap = getSortedNeighborhoodMapFromCache(location);
+    List<Neighborhood> nearestNeighborhoodList = new ArrayList<>();
```

Comment:
Next time, we could use:

Lists.newArrayList(Iterables.limit(sortedNeighborhoodMap.keySet(), n));

Reply:
This is good to know.